### PR TITLE
KIALI-915 Remove Istio infra services by default

### PR DESCRIPTION
**Pull request by jshaughn**
The graph and metrics should not, by default, include request traffic to
the standard istio-system services, like istio-telemetry. This introduces
the includeIstio option, which defaults to false.

Also: remove some dead code re: overview graph

### CODE REVIEW

File: graph/options/options.go [hunk 1]
```diff
@@ -16,8 +16,9 @@ import (
 )
 
 const (
-	AppenderAll  string = "_all_"
-	NamespaceAll string = "all"
+	AppenderAll          string = "_all_"
+	NamespaceAll         string = "all"
+	NamespaceIstioSystem string = "istio-system"
```
Comment (jotak) [inline graph/options/options.go]: I was pretty sure that we had it configured somewhere from config.yaml, but it turns out it's wrong. Except in grafana and jaeger config, but that's not appropriate here.

So, the question is not specific to this PR but more to kiali in general, can we not have istio installed elsewhere than in istio-system?

File: handlers/graph.go [hunk 2]
```diff
@@ -172,15 +172,15 @@ func buildNamespaceTrees(namespace string, o options.Options, client *prometheus
...
-		namespacePattern,          // regex for namespace-constrained service
-		namespacePattern,          // regex for namespace-constrained service
-		"[2345][0-9][0-9]",        // regex for valid response_codes
-		int(o.Duration.Seconds()), // range duration for the query
-		"source_service")          // group by
+		namespacePattern,                // regex for namespace-constrained service
+		namespacePattern,                // regex for namespace-constrained service
+		"[2345][0-9][0-9]",              // regex for valid response_codes
+		int(o.Duration.Seconds()),       // range duration for the query
+		"source_service,source_version") // group by
```
Comment (jotak) [inline handlers/graph.go]: it doesn't looks like it's related to this PR purpose, is it something that was accidentally missing ? (i'd be fine with that, just in order to understand)

File: graph/options/options.go (hunk 1 above)
Comment (jshaughn) [reply graph/options/options.go]: I honestly don't know if Istio can show up in a different namespace that istio-system.  I think at least for now this is a fair assumption.

File: handlers/graph.go (hunk 2 above)
Comment (jshaughn) [reply handlers/graph.go]: You are correct, I removed the assumption that the version would be 'unknown'.  This came up while I was manually testing this PR.

File: graph/options/options.go [hunk 3]
```diff
@@ -60,7 +63,9 @@ func NewOptions(r *http.Request) Options {
 		checkError(err)
 
 		for _, namespace := range namespaces {
-			namespaceNames = append(namespaceNames, namespace.Name)
+			if namespace.Name != NamespaceIstioSystem {
```
Comment (jotak) [inline graph/options/options.go]: Maybe a comment here to say that `includeIstio` flag is not used for graph generation but only for metrics?
Else, I'm pretty sure some day someone will look at this code and ask why this "if" doesn't check `includeIstio` as well :)

Comment (jshaughn) [reply graph/options/options.go]: I added a comment here in a different PR...

**Pull request merged**